reconcile/blackbox_exporter_endpoint_monitoring.py

```diff
@@ -0,0 +1,180 @@
+import logging
+import sys
+from urllib.parse import urlparse
+from typing import Any, Optional
+import json
+from dataclasses import field
+
+from pydantic.dataclasses import dataclass
+
+from reconcile import queries
+from reconcile.utils.openshift_resource import (
+    OpenshiftResource, ResourceInventory)
+from reconcile.utils.defer import defer
+from reconcile.utils.semver_helper import make_semver
+import reconcile.openshift_base as ob
+
+QONTRACT_INTEGRATION = "blackbox-exporter-endpoint-monitoring"
+QONTRACT_INTEGRATION_VERSION = make_semver(0, 1, 0)
+
+LOG = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, eq=True)
+class BlackboxMonitoringProvider:
+
+    module: str
+    namespace: dict[str, Any] = field(compare=False, hash=False)
+    exporterUrl: str
+
+
+@dataclass(frozen=True, eq=True)
+class EndpointMonitoringProvider:
+
+    name: str
+    provider: str
+    description: str
+    timeout: Optional[str] = None
+    checkInterval: Optional[str] = None
+    blackboxExporter: Optional[BlackboxMonitoringProvider] = None
+    metricLabels: Optional[str] = None
+
+    @property
+    def metric_labels(self):
+        return json.loads(self.metricLabels) if self.metricLabels else {}
+
+
+@dataclass
+class Endpoint:
+
+    name: str
+    description: str
+    url: str
+
+    @dataclass
+    class Monitoring:
+
+        provider: EndpointMonitoringProvider
+
+    monitoring: Monitoring
+
+
+def parse_prober_url(url: str) -> dict[str, str]:
+    parsed_url = urlparse(url)
+    if parsed_url.scheme not in ["http", "https"]:
+        raise ValueError(
+            "the prober URL needs to be an http:// or https:// one "
+            f"but is {url}"
+        )
+    data = {
+        "url": parsed_url.netloc,
+        "scheme": parsed_url.scheme
+    }
+    if parsed_url.path:
+        data["path"] = parsed_url.path
+    return data
+
+
+def build_probe(provider: EndpointMonitoringProvider,
+                endpoints: list[Endpoint]) -> Optional[OpenshiftResource]:
+    blackbox_exporter = provider.blackboxExporter
+    if blackbox_exporter:
+        prober_url = parse_prober_url(blackbox_exporter.exporterUrl)
+        body: dict[str, Any] = {
+            "apiVersion": "monitoring.coreos.com/v1",
+            "kind": "Probe",
+            "metadata": {
+                "name": f"endpoint-monitoring-{provider.name}",
+                "namespace": blackbox_exporter.namespace.get("name"),
+                "labels": {
+                    "prometheus": "app-sre"
```

Comment:
tight too much towards app-sre. should we introduce prometheus selector labels in the [EndpointMonitoringProviderBlackboxExporterSettings_v1](https://github.com/app-sre/qontract-schemas/pull/45/files#diff-6bad3cee90319b7494304e3ce4e439992becd03f9d8825eca7f44687f8184ee9R2260) schema?

Reply:
There's only other prometheus that we manage, the prometheus aggregator instance in Quay, and we're not going to use it for this. So I don't think it is worth the effort. If we ever need it, let's add it.